akka-stream/src/main/scala/akka/stream/scaladsl/FlattenStrategy.scala

```diff
@@ -15,7 +20,10 @@ object FlattenStrategy {
    * emitting its elements directly to the output until it completes and then taking the next stream. This has the
    * consequence that if one of the input stream is infinite, no other streams after that will be consumed from.
    */
-  def concat[T]: FlattenStrategy[scaladsl.Source[T, _], T] = Concat[T]()
+  def concat[T]: FlattenStrategy[Source[T, _], T] = Concat[T]()
 
-  private[akka] final case class Concat[T]() extends FlattenStrategy[scaladsl.Source[T, _], T]
+  private[akka] final case class Concat[T]() extends FlattenStrategy[Source[T, _], T] {
+    override def asJava[S]: javadsl.FlattenStrategy[S, T] =
```

Comment:
Not sure what you mean: the variance annotation is only interesting outside of the declaration, on the inside the compiler will make sure that you don’t do anything wrong—meaning if my proposal compiles then it is fine ;-)

Another way to look at this: variance means that a client of this type can see a different type argument, in this case it could be a supertype, but that would also mean that said client will compute a different return type for this method. Whether or not that computation would violate type constraints is automatically verified by the compiler.